lib/net/ldap/connection.rb

```diff
@@ -691,10 +727,19 @@ def delete(args)
     write(request, controls, message_id)
     pdu = queued_read(message_id)
 
-    if !pdu || pdu.app_tag != Net::LDAP::PDU::DeleteResponse
-      raise Net::LDAP::LdapError, "response missing or invalid"
-    end
+    assert_response!(pdu, Net::LDAP::PDU::DeleteResponse)
 
     pdu
   end
+
+  ##
+  # Raise +Net::LDAP::LdapError+ if the response is missing or an unexpected
+  # response tag is returned.
+
+  def assert_response!(pdu, tag)
+    return if pdu && pdu.app_tag == tag
+    raise Net::LDAP::LdapError, 'response missing or invalid'
+  end
+  private :assert_response!
```

Comment:
@jch this is a runtime assertion, not in test. We were already enforcing this assertion, but it was repeated everywhere, hence the abstraction.